src/components/batch_select.vue

```diff
@@ -0,0 +1,185 @@
+<template>
+  <div class="batch-select">
+    <button
+      class="button white-button batch-select-button"
+      @click="d_show_batch_select_modal = true"
+    >
+      Batch Select
+    </button>
+    <div @click.stop>
+      <modal
+        v-if="d_show_batch_select_modal"
+        @close="d_show_batch_select_modal = false"
+        size="large"
+        click_outside_to_close
+      >
+        <div class="modal-header">
+          Select Items ({{ d_selected_items.length }} out of
+          {{ choices.length }} items selected)
+        </div>
+        <input
+          class="input batch-search-field"
+          type="text"
+          placeholder="Enter a name"
+          v-model="d_batch_search_query"
+        />
+        <div>
+          <ul class="batch-select-card-grid">
+            <li
+              class="batch-select-card"
+              :class="{
+                selected: d_selected_items.some((el) => are_items_equal(el, item)),
+              }"
+              v-for="item of batch_filtered_items"
+              :key="item.pk"
+              @click="batch_toggle_select(item)"
+            >
+              {{ item.name || item.pattern }}
```

Comment:
We should be able to use a scoped slot here so that this component is more flexible. Something like: 

```vue
<slot v-bind:item="item"></slot>
```

And then in `suite_settings.vue` something like:
```vue
<batch-select
  v-model="suite.instructor_files_needed"
  :choices="project.instructor_files"
  :are_items_equal="are_files_equal"
  :filter_fn="instructor_file_filter_fn"
  @items_selected="set_instructor_files($event)"
  v-slot:default="{item}"  <!-- NEW -->
>
  {{item.name}}  <!-- use item.name for instructor file, item.pattern for expected student file -->
</batch-select>
```

Reply:
Sorry I missed that earlier